src/pyodide/python-entrypoint-helper.js

```diff
@@ -54,49 +54,65 @@ function initializePackageIndex(pyodide) {
 
 // These packages are currently embedded inside EW and so don't need to
 // be separately installed.
-const EMBEDDED_PYTHON_PACKAGES = new Set([
-  "aiohttp",
-  "aiosignal",
-  "anyio",
-  "async_timeout",
-  "attrs",
-  "certifi",
-  "charset_normalizer",
-  "dataclasses_json",
-  "distro",
-  "fastapi",
-  "frozenlist",
-  "h11",
-  "httpcore",
-  "httpx",
-  "idna",
-  "jsonpatch",
-  "jsonpointer",
-  "langchain",
-  "langchain_community",
-  "langchain_core",
-  "langsmith",
-  "marshmallow",
-  "micropip",
-  "multidict",
-  "mypy_extensions",
-  "numpy",
-  "openai",
-  "packaging",
-  "pydantic",
-  "PyYAML",
-  "requests",
-  "setuptools",
-  "sniffio",
-  "SQLAlchemy",
-  "starlette",
-  "tenacity",
-  "tqdm",
-  "typing_extensions",
-  "typing_inspect",
-  "urllib3",
-  "yarl",
-]);
+// If a version number is specified (i.e. non-empty), it is passed to micropip
+// when running in local dev.
+const EMBEDDED_PYTHON_PACKAGES = {
+  "aiohttp": "3.9.3",
+  "aiosignal": "1.3.1",
+  "anyio": "3.7.1",
+  "async_timeout": "",
+  "attrs": "23.2.0",
+  "certifi": "2024.2.2",
+  "charset_normalizer": "3.3.2",
+  "dataclasses_json": "0.6.4",
+  "distro": "1.9.0",
+  "fastapi": "0.109.2",
+  "frozenlist": "1.4.1",
+  "h11": "0.14.0",
+  "httpcore": "1.0.3",
+  "httpx": "0.26.0",
+  "idna": "3.6",
+  "jsonpatch": "1.33",
+  "jsonpointer": "2.4",
+  "langchain": "0.0.339",
+  "langchain_community": "0.0.20",
+  "langchain_core": "0.1.23",
+  "langsmith": "0.0.87",
+  "marshmallow": "3.20.2",
+  "micropip": "0.6.0",
+  "multidict": "6.0.5",
+  "mypy_extensions": "1.0.0",
+  "numpy": "1.26.4",
+  "openai": "0.28.1",
+  "packaging": "23.2",
+  "pydantic": "2.6.1",
+  "PyYAML": "6.0.1",
+  "requests": "2.31.0",
+  "setuptools": "",
+  "sniffio": "1.3.0",
+  "SQLAlchemy": "2.0.27",
+  "starlette": "0.36.3",
+  "tenacity": "8.2.3",
+  "tqdm": "4.66.2",
+  "typing_extensions": "4.9.0",
+  "typing_inspect": "0.9.0",
+  "urllib3": "2.2.0",
+  "yarl": "1.9.4"
+}
+
+function addPinnedVersionToRequirement(requirement) {
+  if (requirement.includes("==")) {
```

Comment:
I think it would be better to check for presence of the `requirement` against the lock file. That would also catch package names that we don't have / have a typo.

Reply:
@hoodmane that would come after we disable micropip on edgeworker, right?